deepchem/models/tensorgraph/layers.py

```diff
@@ -4303,3 +4303,23 @@ def batch_mat_mult(self, A, B):
     result = tf.matmul(A_reshape, B)
     result = tf.reshape(result, tf.stack([A_shape[0], A_shape[1], axis_2]))
     return result
+
+
+class Hingeloss(Layer):
+
+  def __init__(self, in_layers=None, **kwargs):
+    super(Hingeloss, self).__init__(in_layers, **kwargs)
+    try:
+      self._shape = self.in_layers[1].shape
```

Comment:
This shape is incorrect -- that is why your tests are failing.
https://www.tensorflow.org/api_docs/python/tf/losses/hinge_loss

The shape would be correct if you call
```python
tf.losses.hing_loss(labels=labels, logits=logits, reduction=None)
```
with the reduction=None keyword.

Reply:
that was a doubt, so calling it with reduction=None is not a problem?